cron_scripts/password_expiration_notification.pl

```diff
@@ -0,0 +1,290 @@
+            if ( defined $users_attr->{'zimbraPasswordModifiedTime'} ) {
+                my $zimbraPasswordModifiedTime =
+                  $users_attr->{'zimbraPasswordModifiedTime'};
+                $zimbraPasswordModifiedTime =
+                  substr $zimbraPasswordModifiedTime, 0, 12;
+                my $pass_expiration_year_days_users =
+                  Time::Piece->strptime( $zimbraPasswordModifiedTime,
+                    "%Y%m%d%H%M%S" );
+                my $expire_in_days =
+                  $pass_expiration_year_days -
+                  $pass_expiration_year_days_users->yday;
+                $users_attr->{'expire_in_days'} = $expire_in_days;
+            }
+            push @data, $users_attr;
+        }
+    }
+    else {
+        warn(   "Error: SearchDirectoryRequest "
+              . $soap_api->Error
+              . " via $config->{MAILSTOREHOST} \n" );
+    }
+    if ( scalar @data != 0 ) {
+        my $pm = Parallel::ForkManager->new( $opts{max_proc} );
+      ACCOUNTS:
+        foreach my $acc (@data) {
+            $pm->start and next ACCOUNTS;
+            my $mail           = $acc->{mail};
+            my $mobile         = $acc->{mobile};
+            my $displayName    = $acc->{displayName} || 'User';
+            my $expire_in_days = $acc->{'expire_in_days'};
+            if ( $mail ) {
+                my $mail_content = $config->{mail_content};
+                $mail_content =~ s/{displayName}/$displayName/g;
+                $mail_content =~ s/{numDays}/$expire_in_days/g;
+                my $mail_data = {
+                    'to' => $mail,
+                    'subject' =>
+"Your Mail access password expire in $acc->{'expire_in_days'} days",
+                    'body' => $mail_content
+                };
+                if ( my $sendmail =
+                    eval { $soap_api->sendmessage($mail_data) } )
+                {
+                    warn("DEBUG: Mail sent to $mail\n") if ($Debug);
+                }
+                else {
+                    warn( "Error: for $mail on sending mail. Message: "
+                          . ZCS::CustomAPI->Error );
+                }
+            }
+            if ( $mobile && $SendSMS ) {
+                my $sms_content = $config->{sms_content};
+                $sms_content =~ s/{displayName}/$displayName/g;
+                $sms_content =~ s/{numDays}/$expire_in_days/g;
+                my $sms_url =
+"$config->{SMSHOST}?username=$config->{SMSUSER}&pin=$config->{SMSPIN}&mnumber=$mobile&message=$sms_content";
+
+#https://hostname/HttpLink?username=zimbra.sms&pin=redacted&message=hifromnic&mnumber=919922429908&signature=NICSMS
+
+                my $sms_res = LWP::UserAgent->new->get($sms_url);
+
+                #{ "_rc": 500, "_msg": "SMS Sent Successfully" }
+                if ( $sms_res->{'_rc'} != 000 ) {
+                    warn(
+"Error: for $mail on sending SMS. Code: $sms_res->{'_rc'} Message: $sms_res->{'_msg'}"
```

Comment:
nit: This wording is rough.  I would also update wording similar to suggestion on mail above.